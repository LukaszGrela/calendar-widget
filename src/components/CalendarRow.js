import React from 'react';
import PropTypes from 'prop-types';
import CalendarDay from './CalendarDay';
import { momentObj } from 'react-moment-proptypes';
import { noop } from '../utils/helpers';

class CalendarRow extends React.Component {

    renderCalendarDays = () => {
        const { days, now, current, dayClicked = noop } = this.props;
        return days.map((day, index) => {

            const spill = typeof date === 'string' || !current ? '' :
                (day.month() !== current.month() ? ' spill' : '');
            const today = typeof date === 'string' || !now ? '' :
                (day.month() === now.month() &&
                    day.date() === now.date() &&
                    day.year() === now.year() ? ' today' : '');

            return (
                <CalendarDay
                    key={day}
                    className={`day-${index}${spill}${today}`}
                    date={day}
                    handleClick={dayClicked}
                />
            );
        });
    }

    render = () => {
        const { className } = this.props;
        return (
            <div className={'row' + (className ? ' ' + className : '')}>
                {
                    this.renderCalendarDays()
                }
            </div>
        );
    }
}
CalendarRow.propTypes = {
    days: PropTypes.array.isRequired,
    now: momentObj,
    current: momentObj,
    dayClicked: PropTypes.func,
    className: PropTypes.string
};

export default CalendarRow;