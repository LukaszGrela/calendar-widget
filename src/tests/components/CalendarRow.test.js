import React from 'react';
import { shallow } from 'enzyme';
import moment from 'moment';

import CalendarRow from '../../components/CalendarRow';
import { calendarDates } from '../../utils/helpers';

let wrapper, days, now, current;
beforeEach(() => {
    now = moment();
    current = moment().add(1, 'days');
    days = calendarDates(now);
    wrapper = shallow(<CalendarRow
        days={days[0]}
    />)
})
test('should render correctly', () => {
    // labels
    expect(wrapper).toMatchSnapshot();
});
test('should render correctly week labels', () => {
    wrapper.setProps({
        days: ["mon", "tue", "wed", "thu", "fri", "sat", "sun"]
    });
    expect(wrapper).toMatchSnapshot();
});

test('should render correctly with given className', () => {
    wrapper.setProps({
        className: 'my-class'
    })
    expect(wrapper).toMatchSnapshot();
});


test('should render correctly apply spill and today class', () => {
    wrapper.setProps({
        now: now,
        current: current
    });
    expect(wrapper).toMatchSnapshot();
});
