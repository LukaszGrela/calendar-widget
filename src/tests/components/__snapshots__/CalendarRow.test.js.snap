// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="row"
>
  <CalendarDay
    className="day-0"
    date={"1969-12-28T00:00:00.000Z"}
    handleClick={[Function]}
    key="-345600000"
  />
  <CalendarDay
    className="day-1"
    date={"1969-12-29T00:00:00.000Z"}
    handleClick={[Function]}
    key="-259200000"
  />
  <CalendarDay
    className="day-2"
    date={"1969-12-30T00:00:00.000Z"}
    handleClick={[Function]}
    key="-172800000"
  />
  <CalendarDay
    className="day-3"
    date={"1969-12-31T00:00:00.000Z"}
    handleClick={[Function]}
    key="-86400000"
  />
  <CalendarDay
    className="day-4"
    date={"1970-01-01T00:00:00.000Z"}
    handleClick={[Function]}
    key="0"
  />
  <CalendarDay
    className="day-5"
    date={"1970-01-02T00:00:00.000Z"}
    handleClick={[Function]}
    key="86400000"
  />
  <CalendarDay
    className="day-6"
    date={"1970-01-03T00:00:00.000Z"}
    handleClick={[Function]}
    key="172800000"
  />
</div>
`;

exports[`should render correctly apply spill and today class 1`] = `
<div
  className="row"
>
  <CalendarDay
    className="day-0 spill"
    date={"1969-12-28T00:00:00.000Z"}
    handleClick={[Function]}
    key="-345600000"
  />
  <CalendarDay
    className="day-1 spill"
    date={"1969-12-29T00:00:00.000Z"}
    handleClick={[Function]}
    key="-259200000"
  />
  <CalendarDay
    className="day-2 spill"
    date={"1969-12-30T00:00:00.000Z"}
    handleClick={[Function]}
    key="-172800000"
  />
  <CalendarDay
    className="day-3 spill"
    date={"1969-12-31T00:00:00.000Z"}
    handleClick={[Function]}
    key="-86400000"
  />
  <CalendarDay
    className="day-4 today"
    date={"1970-01-01T00:00:00.000Z"}
    handleClick={[Function]}
    key="0"
  />
  <CalendarDay
    className="day-5"
    date={"1970-01-02T00:00:00.000Z"}
    handleClick={[Function]}
    key="86400000"
  />
  <CalendarDay
    className="day-6"
    date={"1970-01-03T00:00:00.000Z"}
    handleClick={[Function]}
    key="172800000"
  />
</div>
`;

exports[`should render correctly week labels 1`] = `
<div
  className="row"
>
  <CalendarDay
    className="day-0"
    date="mon"
    handleClick={[Function]}
    key="mon"
  />
  <CalendarDay
    className="day-1"
    date="tue"
    handleClick={[Function]}
    key="tue"
  />
  <CalendarDay
    className="day-2"
    date="wed"
    handleClick={[Function]}
    key="wed"
  />
  <CalendarDay
    className="day-3"
    date="thu"
    handleClick={[Function]}
    key="thu"
  />
  <CalendarDay
    className="day-4"
    date="fri"
    handleClick={[Function]}
    key="fri"
  />
  <CalendarDay
    className="day-5"
    date="sat"
    handleClick={[Function]}
    key="sat"
  />
  <CalendarDay
    className="day-6"
    date="sun"
    handleClick={[Function]}
    key="sun"
  />
</div>
`;

exports[`should render correctly with given className 1`] = `
<div
  className="row my-class"
>
  <CalendarDay
    className="day-0"
    date={"1969-12-28T00:00:00.000Z"}
    handleClick={[Function]}
    key="-345600000"
  />
  <CalendarDay
    className="day-1"
    date={"1969-12-29T00:00:00.000Z"}
    handleClick={[Function]}
    key="-259200000"
  />
  <CalendarDay
    className="day-2"
    date={"1969-12-30T00:00:00.000Z"}
    handleClick={[Function]}
    key="-172800000"
  />
  <CalendarDay
    className="day-3"
    date={"1969-12-31T00:00:00.000Z"}
    handleClick={[Function]}
    key="-86400000"
  />
  <CalendarDay
    className="day-4"
    date={"1970-01-01T00:00:00.000Z"}
    handleClick={[Function]}
    key="0"
  />
  <CalendarDay
    className="day-5"
    date={"1970-01-02T00:00:00.000Z"}
    handleClick={[Function]}
    key="86400000"
  />
  <CalendarDay
    className="day-6"
    date={"1970-01-03T00:00:00.000Z"}
    handleClick={[Function]}
    key="172800000"
  />
</div>
`;
